PYME/recipes/pointcloud.py

```diff
@@ -251,20 +251,31 @@ class Ripleys(ModuleBase):
     inputPositions = Input('input')
     inputMask = Input('')
     outputName = Output('ripleys')
-    normalization = Enum(['K', 'L'])
+    normalization = Enum(['K', 'L', 'H'])
     nbins = Int(50)
     binSize = Float(50.)
     sampling = Float(5.)
     threaded = Bool(False)
+    dimension = Enum(['2D', '3D'])
     
     def execute(self, namespace):
         from PYME.Analysis.points import ripleys
         from PYME.IO import MetaDataHandler
         
         points_real = namespace[self.inputPositions]
         mask = namespace.get(self.inputMask, None)
-        
-        three_d = np.count_nonzero(points_real['z']) > 0
+
+        # three_d = np.count_nonzero(points_real['z']) > 0
+        if self.dimension == '3D':
+            if np.count_nonzero(points_real['z']) == 0:
+                raise RuntimeError('Need a 3D dataset')
+            if mask.data.shape[2] < 2:
+                raise RuntimeError('Need a 3D mask to run in 3D. Generate a 3D mask or select 2D.')
+            three_d = True
+        if self.dimension == '2D':
```

Comment:
```suggestion
        else:
```

There are only two options, and we can only be one or the other. Something like
```python
if (dim=='3D'):
    ...
elif (dim=='2D'):
    ...
else:
   raise (InvalidDimChoice)
```
would also be acceptable, but overkill as the `Enum` trait does the validation for you.

Independent `if` statements implies that they are not mutually exclusive - i.e. both could evaluate true

Reply:
Yup, strange. I thought I'd written this as an `if`/`elif`. Thanks for catching that!